### DISCUSSION TITLE
Is it possible to generate operations based on the schema?

### DISCUSSION THREAD

**Discussion by goldenpath-ai** [dotansimha/graphql-code-generator#6295] [Q&A] [answered]
First of all, thanks for this tool, it's awesome and makes my life so much easier on both Backend and Frontend.

But is there a way to automatically generate the operations based on my schema?

In order for the useXMutation or useXQuery React hook to get generated, I have to create the operation by hand.

```
mutation register($input: RegisterInput!) {
  register(input: $input) {
    success
    errors {
      field
      message
    }
  }
}
```

Which is no big deal, but it's but it'd be awesome if it could somehow be pre-generated, maybe with all the possible fields used by default.

Is it possible to do this, with this or any other tool?

**Comment by ardatan** [comment, marked as answer]
See https://github.com/dotansimha/graphql-code-generator/discussions/2349
